### PR TITLE
Pass RollupParams as an argument in `prover-client`

### PR DESCRIPTION
**Pull request by prajwolrg**
## Description

`RollupParams` was hardcoded in a file inside `prover-client`. This PR fixes that by passing it as an argument.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactor
- [ ] New or updated tests
- [x] Dependency Update

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code where necessary.
- [x] I have updated the documentation if needed.
- [x] My changes do not introduce new warnings.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [x] New and existing tests pass with my changes.

### CODE REVIEW

File: bin/prover-client/src/main.rs [hunk 1]
```diff
@@ -50,7 +50,16 @@ async fn main() {
     )
     .expect("failed to connect to the btc client");
 
-    let operator = Arc::new(ProofOperator::init(btc_client, el_client, cl_client));
+    let rollup_params = args
+        .resolve_and_validate_rollup_params()
+        .expect("failed to resolve rollup params");
```
Comment (delbonis) [inline bin/prover-client/src/main.rs:55]: Should have better error handling here.

File: bin/prover-client/src/args.rs [hunk 2]
```diff
@@ -173,4 +178,13 @@ impl Args {
...
         workers
     }
+
+    /// Resolves the rollup params file to use, from a path, and validates
+    /// it to ensure it passes sanity checks.
+    pub fn resolve_and_validate_rollup_params(&self) -> anyhow::Result<RollupParams> {
+        let json = fs::read_to_string(&self.rollup_params)?;
+        let rollup_params = from_str::<RollupParams>(&json)?;
+        rollup_params.check_well_formed()?;
+        Ok(rollup_params)
```
Comment (delbonis) [inline bin/prover-client/src/args.rs:188]: Isn't file on disk is the whole `Params` structure?  We can extract the inner `RollupParams` from there.

Comment (prajwolrg) [reply bin/prover-client/src/args.rs:188]: The file on disk only requires the RollupParams structure. Only RollupParams is set during the functional tests as well.

**Review by evgenyzdanovich — APPROVED** [review]
Nice work, LGTM!

**Review by storopoli — APPROVED** [review]
ACK 505ee173efc1476f5df4c5ebe6d2aebf54b71a00

**Review by delbonis — APPROVED** [review]

**Pull request merged**